### PR TITLE
[BTCV2-1774]Feature/upgrade to rn64.2

### PR DESCRIPTION
**Pull request by mlcbest**
## What does this PR do?
Upgrade react-native to 64.2
Integrate flipper
Update splashscreen show/hide
Fix Android splash screen not hide
add types for sagas as latest typescript not allow yield to be any by default
fixed jest to handle upgraded react-native

## Todos:

- [+] Checked on iOS
- [+] Checked on Android

### CODE REVIEW

File: BlueElectrum.js [hunk 1]
```diff
@@ -153,11 +153,13 @@ module.exports.getTransactionsByAddress = async function(address) {
 
 module.exports.ping = async function() {
   try {
+    await wait(200); // TODO: For some reason sometimes Android not not load (infinity internet connection error)
```
Comment (PanRadaCloudbest) [inline BlueElectrum.js:156]: its not to simple hack :) ? need test ?

File: android/app/build.gradle [hunk 2]
```diff
@@ -24,7 +24,7 @@ apply from: project(':react-native-config').projectDir.getPath() + "/dotenv.grad
 project.ext.react = [
     entryFile: "index.js",
     devDisabledInStaging: true,
-    enableHermes: false
+    enableHermes: true
```
Comment (PanRadaCloudbest) [inline android/app/build.gradle:27]: for some reason that was switch off - need find way

File: android/app/build.gradle [hunk 3]
```diff
@@ -135,17 +136,29 @@ android {
         variant.outputs.each { output ->
             // For each separate APK per architecture, set a unique version code as described here:
             // https://developer.android.com/studio/build/configure-apk-splits.html
+            // Example: versionCode 1 will generate 1001 for armeabi-v7a, 1002 for x86, etc.
             def versionCodes = ["armeabi-v7a": 1, "x86": 2, "arm64-v8a": 3, "x86_64": 4]
             def abi = output.getFilter(OutputFile.ABI)
             if (abi != null) {  // null for the universal-debug, universal-release variants
                 output.versionCodeOverride =
-                        versionCodes.get(abi) * 1048576 + defaultConfig.versionCode
+                        defaultConfig.versionCode * 1000 + versionCodes.get(abi)
```
Comment (PanRadaCloudbest) [inline android/app/build.gradle:144]: it will be pass in bitrise ?

File: android/app/build.gradle [hunk 4]
```diff
@@ -135,17 +136,29 @@ android {
...
         }
     }
 }
 
+repositories {
+  mavenCentral()
+}
+
 dependencies {
+    debugImplementation("com.facebook.flipper:flipper:${FLIPPER_VERSION}") {
```
Comment (PanRadaCloudbest) [inline android/app/build.gradle:155]: we are sure we wanna flipper ? is million reason for not - builds in bitrise and others

File: android/app/src/main/res/values/colors.xml [hunk 5]
```diff
@@ -1,4 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources>
     <color name="bootsplash_background">#FFFFFF</color>
+    <color name="primary_dark">#000000</color>
```
Comment (PanRadaCloudbest) [inline android/app/src/main/res/values/colors.xml:4]: why ?

File: android/app/src/main/res/values/styles.xml [hunk 6]
```diff
@@ -1,13 +1,19 @@
 <resources>
 
     <!-- Base application theme. -->
-    <style name="AppTheme" parent="Theme.AppCompat.Light.NoActionBar">
+    <style name="AppTheme" parent="Theme.AppCompat.DayNight.NoActionBar">
```
Comment (PanRadaCloudbest) [inline android/app/src/main/res/values/styles.xml:4]: why ?

File: android/app/src/main/res/values/styles.xml [hunk 7]
```diff
@@ -1,13 +1,19 @@
...
         <!-- Customize your theme here. -->
         <item name="android:editTextStyle">@style/AppEditTextStyle</item>
         <item name="editTextStyle">@style/AppEditTextStyle</item> <!-- For compatibility with the support library -->
         <item name="android:editTextBackground">@android:color/transparent</item>
         <item name="android:textColor">#000000</item>
     </style>
+    <!-- Add the following lines (BootTheme should inherit from AppTheme) -->
```
Comment (PanRadaCloudbest) [inline android/app/src/main/res/values/styles.xml:11]: same as on top why ?

File: android/build.gradle [hunk 8]
```diff
@@ -2,21 +2,22 @@
...
         kotlinVersion = '1.4.10'
+        ndkVersion = "20.1.5948944"
     }
     repositories {
         google()
         jcenter()
     }
     dependencies {
-        classpath("com.android.tools.build:gradle:3.5.3")
+        classpath("com.android.tools.build:gradle:4.1.0")
```
Comment (PanRadaCloudbest) [inline android/build.gradle:20]: need changes in bitrise ?

File: ios/GoldWallet-beta.plist [hunk 9]
```diff
@@ -57,8 +57,6 @@
 	<true/>
 	<key>NSAppTransportSecurity</key>
 	<dict>
-		<key>NSAllowsArbitraryLoads</key>
```
Comment (PanRadaCloudbest) [inline ios/GoldWallet-beta.plist:60]: will not impact network rulez in app like deeplinking or others secure ?

File: ios/GoldWallet/AppDelegate.m [hunk 10]
```diff
@@ -26,7 +26,11 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
   RCTRootView *rootView = [[RCTRootView alloc] initWithBridge:bridge
                                                       moduleName:[ReactNativeConfig envFor:@"APPLICATION_NAME"]
                                                initialProperties:nil];
-  rootView.backgroundColor = [UIColor whiteColor];
+  if (@available(iOS 13.0, *)) {
+      rootView.backgroundColor = [UIColor systemBackgroundColor];
```
Comment (PanRadaCloudbest) [inline ios/GoldWallet/AppDelegate.m:30]: as i understood it is gate for dark mode in app ?

File: ios/GoldWallet/Info.plist [hunk 11]
```diff
@@ -57,8 +57,6 @@
 	<true/>
 	<key>NSAppTransportSecurity</key>
 	<dict>
-		<key>NSAllowsArbitraryLoads</key>
```
Comment (PanRadaCloudbest) [inline ios/GoldWallet/Info.plist:60]: same as top

File: ios/GoldWalletDev-Info.plist [hunk 12]
```diff
@@ -57,8 +57,6 @@
 	<true/>
 	<key>NSAppTransportSecurity</key>
 	<dict>
-		<key>NSAllowsArbitraryLoads</key>
```
Comment (PanRadaCloudbest) [inline ios/GoldWalletDev-Info.plist:60]: same as top

File: ios/GoldWalletStage-Info.plist [hunk 13]
```diff
@@ -57,8 +57,6 @@
 	<true/>
 	<key>NSAppTransportSecurity</key>
 	<dict>
-		<key>NSAllowsArbitraryLoads</key>
```
Comment (PanRadaCloudbest) [inline ios/GoldWalletStage-Info.plist:60]: same as top

File: ios/Podfile [hunk 14]
```diff
@@ -57,4 +60,14 @@ end
 target 'GoldWallet Dev' do
   sharedPods
   pod 'react-native-config', :path => '../node_modules/react-native-config'
+
+    # Enables Flipper.
+  #
+  # Note that if you have use_frameworks! enabled, Flipper will not work and
+  # you should disable the next line.
+  use_flipper!()
```
Comment (PanRadaCloudbest) [inline ios/Podfile:68]: yee same as top we need this flipper ? suspect lots problems in ours CI/CD

File: src/helpers/sagas.ts [hunk 15]
```diff
@@ -10,8 +10,8 @@ export function takeLatestPerKey(
     const tasks = {};
 
     while (true) {
-      const action = yield take(patternOrChannel);
-      const key = yield call(keySelector, action);
+      const action: any = yield take(patternOrChannel);
```
Comment (PanRadaCloudbest) [inline src/helpers/sagas.ts:13]: ? why any?

File: src/state/authentication/sagas.ts [hunk 16]
```diff
@@ -1,6 +1,6 @@
 import { takeLatest, takeEvery, put, call } from 'redux-saga/effects';
```
Comment (PanRadaCloudbest) [inline src/state/authentication/sagas.ts:1]: to this and all sagas files - is any particular reason u tach this files in this commit ? if it is connect with some internet connection reason will be better have this in other PR pls,

**Review by PanRadaCloudbest — APPROVED** [review]

**Review by PanRadaCloudbest — COMMENTED** [review]
Look ok bitrise build is ok for stage and dev hope ale will correct

File: BlueElectrum.js (hunk 1 above)
Comment (mlcbest) [reply BlueElectrum.js:156]: Not sure about real device but for me using simulator it helped, both dev and prod versions. It doesn't need testing as it used only in saga and create simple delay that will not cause any issues

File: android/app/build.gradle (hunk 2 above)
Comment (mlcbest) [reply android/app/build.gradle:27]: It was disabled As it didn't work with RN 62 fir ios

File: android/app/build.gradle (hunk 3 above)
Comment (mlcbest) [reply android/app/build.gradle:144]: It passed

File: android/app/build.gradle (hunk 4 above)
Comment (mlcbest) [reply android/app/build.gradle:155]: Flipper is nice tool to have and it not bundled in prod version

File: android/app/src/main/res/values/colors.xml (hunk 5 above)
Comment (mlcbest) [reply android/app/src/main/res/values/colors.xml:4]: Was one of upgrade of RN requirements

File: android/app/src/main/res/values/styles.xml (hunk 6 above)
Comment (mlcbest) [reply android/app/src/main/res/values/styles.xml:4]: Another requirement for upgrading

File: android/app/src/main/res/values/styles.xml (hunk 7 above)
Comment (mlcbest) [reply android/app/src/main/res/values/styles.xml:11]: Another requirement for upgrading

File: android/build.gradle (hunk 8 above)
Comment (mlcbest) [reply android/build.gradle:20]: Bitrise is ok, I did all changed it required

File: ios/GoldWallet-beta.plist (hunk 9 above)
Comment (mlcbest) [reply ios/GoldWallet-beta.plist:60]: Nope, it is enabled by default so not restrictions will be applied

File: ios/GoldWallet/AppDelegate.m (hunk 10 above)
Comment (mlcbest) [reply ios/GoldWallet/AppDelegate.m:30]: yes, but we have Light theme forced so it ill not be applied, but required for RN upgrade

File: ios/GoldWallet/Info.plist (hunk 11 above)
Comment (mlcbest) [reply ios/GoldWallet/Info.plist:60]: answered

File: ios/Podfile (hunk 14 above)
Comment (mlcbest) [reply ios/Podfile:68]: answered above

File: src/helpers/sagas.ts (hunk 15 above)
Comment (mlcbest) [reply src/helpers/sagas.ts:13]: fixed

File: src/state/authentication/sagas.ts (hunk 16 above)
Comment (mlcbest) [reply src/state/authentication/sagas.ts:1]: TS fixes were required

**Review by aleparty — APPROVED** [review]

**Pull request merged**